hathor/p2p/manager.py

```diff
@@ -184,23 +177,57 @@ def __init__(self,
         self.whitelist_only = whitelist_only
 
-        self.enable_sync_v1 = enable_sync_v1
-        self.enable_sync_v2 = enable_sync_v2
-
         # Timestamp when the last discovery ran
         self._last_discovery: float = 0.
 
         # sync-manager factories
         self._sync_factories = {}
-        if enable_sync_v1:
-            self._sync_factories[SyncVersion.V1_1] = SyncV11Factory(self)
-        if enable_sync_v2:
-            self._sync_factories[SyncVersion.V2] = SyncV2Factory(self)
+        self._enabled_sync_versions = set()
+
+    def add_sync_factory(self, sync_version: SyncVersion, sync_factory: SyncAgentFactory) -> None:
+        """Add factory for the given sync version, must use a sync version that does not already exist."""
+        # XXX: to allow code in `set_manager` to safely use the the available sync versions, we add this restriction:
+        assert self.manager is None, 'Cannot modify sync factories after a manager is set'
+        if sync_version in self._sync_factories:
+            raise ValueError('sync version already exists')
+        self._sync_factories[sync_version] = sync_factory
+
+    def get_available_sync_versions(self) -> set[SyncVersion]:
+        """What sync versions the manager is capable of using, they are not necessarily enabled."""
+        return set(self._sync_factories.keys())
+
+    def is_sync_version_available(self, sync_version: SyncVersion) -> bool:
+        """Whether the given sync version is available for use, is not necessarily enabled."""
+        return sync_version in self._sync_factories
+
+    def get_enabled_sync_versions(self) -> set[SyncVersion]:
+        """What sync versions are enabled for use, it is necessarily a subset of the available versions."""
+        return self._enabled_sync_versions.copy()
+
+    def is_sync_version_enabled(self, sync_version: SyncVersion) -> bool:
+        """Whether the given sync version is enabled for use, being enabled implies being available."""
+        return sync_version in self._enabled_sync_versions
+
+    def enable_sync_version(self, sync_version: SyncVersion) -> None:
+        """Enable using the given sync version on new connections, it must be available before being enabled."""
+        assert sync_version in self._sync_factories
+        if sync_version in self._enabled_sync_versions:
+            self.log.info('tried to enable a sync verison that was already enabled, nothing to do')
+            return
+        self._enabled_sync_versions.add(sync_version)
+
+    def disable_sync_version(self, sync_version: SyncVersion) -> None:
+        """Disable using the given sync version, closes connections using this sync version."""
+        if sync_version not in self._enabled_sync_versions:
+            self.log.info('tried to disable a sync verison that was already disabled, nothing to do')
+            return
+        # TODO: close connections using the given sync version
```

Comment:
I don't think we should do it. Maybe create a separate method that closes all connections using a specific version.

Reply:
I agree, I removed this comment and amended the docstring.

Reply:
Thanks!